napari/_vispy/overlays/brush_circle.py

```diff
@@ -0,0 +1,75 @@
+from vispy.scene.visuals import Compound, Ellipse
+
+from napari._vispy.overlays.base import ViewerOverlayMixin, VispyCanvasOverlay
+
+
+class VispyBrushCircleOverlay(ViewerOverlayMixin, VispyCanvasOverlay):
+    def __init__(self, *, viewer, overlay, parent=None):
+        self._white_circle = Ellipse(
+            center=(0, 0),
+            color=(0, 0, 0, 0.0),
+            border_color='white',
+            border_method='agg',
+        )
+        self._black_circle = Ellipse(
+            center=(0, 0),
+            color=(0, 0, 0, 0.0),
+            border_color='black',
+            border_method='agg',
+        )
+
+        super().__init__(
+            node=Compound([self._white_circle, self._black_circle]),
+            viewer=viewer,
+            overlay=overlay,
+            parent=parent,
+        )
+
+        self._last_mouse_pos = None
+
+        self.overlay.events.size.connect(self._on_size_change)
+        self.node.events.canvas_change.connect(self._on_canvas_change)
+        self.viewer.events.mouse_over_canvas.connect(
+            self._on_mouse_over_canvas
+        )
+        # no need to connect position, since that's in the base classes of CanvasOverlay
+
+        self.reset()
+
+    def _on_position_change(self, event=None):
+        self._set_position(self.overlay.position)
+
+    def _on_size_change(self, event=None):
+        self._white_circle.radius = self.overlay.size / 2
+        self._black_circle.radius = self._white_circle.radius - 1
+
+    def _on_visible_change(self):
+        if self._last_mouse_pos is not None:
+            self._set_position(self._last_mouse_pos)
+        if not self.viewer.mouse_over_canvas and self.overlay.visible:
+            # if the mouse is not over canvas, don't make the node visible
+            return
+        super()._on_visible_change()
+
+    def _on_mouse_move(self, event):
+        self._last_mouse_pos = event.pos
+        if self.overlay.visible:
+            self.overlay.position = event.pos.tolist()
```

Comment:
```suggestion
    def _on_mouse_move(self, event=None):
        pos = event.pose.tolist() if event is not None else (-1000, -1000)
        self._last_mouse_pos = pos
        self.overlay.position = pos
```

Reply:
I don't like that change. Why should it take event=None? It also makes the code less readable and efficient. It performs two more operations every mouse move not considering the status of the overlay even when it is disabled (additional operation `tolist()` and comparison with `None`)

Having `if self.overlay.visible` is very important, without it your code renders the overlay even when it is disabled. Updating position triggers the vispy update even when the node is not visible.

Reply:
Why it should take the position `(-1000, -1000)` when the event is None is also not clear for me, it doesn't help with any issue (including the discussed about with `_on_mouse_over_canvas`), just adds some magic constants to the code.